src/tonic/api/points_api.rs

```diff
@@ -128,6 +419,40 @@ impl From<GeoPayload> for PayloadInterface {
     }
 }
 
+impl TryFrom<GeoBoundingBox> for segment::types::GeoBoundingBox {
+    type Error = Status;
+
+    fn try_from(value: GeoBoundingBox) -> Result<Self, Self::Error> {
+        match value {
+            GeoBoundingBox {
+                top_left: Some(t),
+                bottom_right: Some(b),
+            } => Ok(Self {
+                top_left: t.into(),
+                bottom_right: b.into(),
+            }),
+            _ => Err(Status::failed_precondition("Malformed GeoBoundingBox type")),
```

Comment:
Also `Status::invalid_argument `

Reply:
I took the liberty to change them all https://github.com/qdrant/qdrant/pull/264/commits/f93b6dd59e079a88bedb70e43f58a94a6ebd6c5d